ios/RNFirebase/notifications/RNFirebaseNotifications.m

```diff
@@ -95,6 +98,15 @@ - (void)didReceiveLocalNotification:(nonnull UILocalNotification *)localNotifica
     }
 }
 
+RCT_EXPORT_METHOD(complete:(NSString*)handlerKey fetchResult:(UIBackgroundFetchResult)fetchResult) {
+    void (^completionHandler)(UIBackgroundFetchResult) = completionHandlers[handlerKey];
+    completionHandlers[handlerKey] = nil;
+
+    if(completionHandler != nil) {
+        completionHandler(fetchResult);
```

Comment:
`completionHandler` should be removed from the `completionHandlers` on consumption?

Reply:
This happens on line 103. We do it before calling the `completionHandler` as the OS may suspend execution when `completionHandler` is called.

Reply:
Oh, I meant `[completionHandlers removeObjectForKey:handlerKey];` specifically, setting to `nil` still keeps the key + a nil object in the dictionary afaik?

Reply:
When I was debugging this I checked and `completionHandlers[handlerKey] = nil` definitely resulted in the key also being removed. Happy to change to be sure though if you'd prefer `[completionHandlers removeObjectForKey:handlerKey]`?

Reply:
If that's the case then as-is is fine - my obj-c some days is 💩